### PR TITLE
Header-Footer

### PR DESCRIPTION
**Pull request by Vagyasri**
### In this project, we have basically retrieved data from a given API and implemented them as follows:

1.  We have Created two Interfaces:
- The home page
- The comments popup

      **Home page**
      
      - When the page loads, the web app retrieves data from the selected API and shows the list of items on the screen
      - When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated
      - When the user clicks on the "Comments" button, the Comments popup appears
      
      **Comments popup**
      
      - When the popup loads, the web app retrieves data from the selected API and shows details about the selected item
      - When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated

2. We have counters in all the interfaces that show:

- The number of items (home)
- The number of comments (comments popup)

Presentation URL: https://drive.google.com/file/d/1nJuMiIlEN3Jh_K2ymww-xDE2vFgc8687/view?usp=sharing

### CODE REVIEW

File: src/newComment.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const finalUrl = await fetch(`${commentUrl}${getPath}`);
+  const finalResult = await finalUrl.json();
+
+  finalResult.forEach((result) => {
+    const commentCount = document.querySelector('.comments-count');
+    const commentList = document.querySelector('.comment-list');
+    const listContent = document.createElement('li');
+    listContent.classList = 'list-content';
+    listContent.innerText = `${result.creation_date} ${result.username}: ${result.comment}`;
+    commentCount.innerText = `(${finalResult.length} Comments)`;
```
Comment (sinansevgi) [inline src/newComment.js:17]: - [x] For this project, you should have a counter function to count comments. Looks like you are directly printing the array length.

File: src/displayImage.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  accessComment();
+  await likesDisplay();
+};
+
+const totalItems = (arg) => {
+  let displayLength = '';
+  displayLength += `
+  <h3>Cuisine ${arg.length}</h3>`;
+  itemList.innerHTML = displayLength;
+};
```
Comment (sinansevgi) [inline src/displayImage.js:49]: - [x] For this project, you should have a counter function to count items. Looks like you are directly printing the array length.

You can easily implement a function that takes an array as an argument and return its length.

File: src/_mocks_/comments.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    new Array(18).fill({
+      comment: 'Indian Cuisine',
+      creation_date: '2021-10-23',
+      username: 'Lucky',
+    }),
+  );
+  return result;
+};
+
+export default commentsCounter;
```
Comment (sinansevgi) [inline src/_mocks_/comments.js:12]: - [x] You should test the counter function that you have used in your application. You have created a new function in the _mocks_ file that has nothing to do with your actual application. You should use mocks files to imitate remote dependencies (API calls, databases, files), not for parts of your application.  Please implement a counter function for your app and test it properly.

File: src/_mocks_/items.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    new Array(14).fill({
+      idMeal: '52959',
+      strMeal: 'Mock for meal counter',
+      strMealThumb: 'https://themealdb.com/api/json/v1/1/categories.php',
+    }),
+  );
+  return result;
+};
+
+export default mealsMock;
```
Comment (sinansevgi) [inline src/_mocks_/items.js:12]: - [x] My comment about the comment counter applies to this counter also.

File: src/displayImage.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        likeVariable = likes.likes;
+      }
+    });
+    display += `   
+      <li id="${element.idCategory}">
+      <img src="${element.strCategoryThumb}" alt="Food">
+      <div class="name">
+      <h4>${element.strCategory}</h4>
+      <i data-id="${element.idCategory}" class="far fa-heart like"></i>
+      <span id="like${element.idCategory}" class="likes-count">${likeVariable} Likes</span>
```
Comment (sinansevgi) [inline src/displayImage.js:33]: - [x] Like counts should get updated automatically when I open the page. (Try to clean local storage and refresh the page) 
Screenshot 2021-10-23 at 21-17-54 Meal

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @Vagyasri and @Ugboaja-Uchechi , 

Good job so far!
Great presentation :+1: 
There are some issues that you still need to work on to go further but you are almost there!


 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/_test_/comments.test.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import commentsCounter from '../_mocks_/comments.js';
```
Comment (rukundoeric) [inline src/_test_/comments.test.js:1]: - [ ] <b>REQUIRED:</b> As the previous review said, the functions you are supposed to test is the real function you are using, not the mocked one. remember, mocking is about the data, not the code.

File: src/_test_/items.test.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import mealsMock from '../_mocks_/items.js';
```
Comment (rukundoeric) [inline src/_test_/items.test.js:5]: - [ ] <b>REQUIRED:</b> The same thing applies here, the functions you are supposed to test is the real function you are using, not the mocked one. remember, mocking is about the data, not the code.

File: src/displayImage.js (hunk 5 above)
Comment (rukundoeric) [inline src/displayImage.js:33]: - [ ] <b>REQUIRED:</b> As the previous reviewer said, the likes counter is showing `0` when you visit the page and get updated only after you click on the like. that's not how it should be. Kindly fix it.

File: src/index.html [hunk 8]
```diff
@@ -4,8 +4,33 @@
...
+  </header>
+  <div class="item-list">
+
+  </div>
+  <main class="main-container">
+    <ul class="meal-list">
+    </ul>
+  </main>
+  <div class="popup-wrapper">
+    <div class="comment-popup"></div>
```
Comment (rukundoeric) [inline src/index.html:30]: - [ ] <b>REQUIRED:</b>  When I add a new comment, your function is displaying a new list with the previous comment duplicated. kindly fix it.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @Vagyasri, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**